stl/src/vector_algorithms.cpp

```diff
@@ -451,6 +464,526 @@ __declspec(noalias) void __cdecl __std_reverse_copy_trivially_copyable_8(
+        return _mm_max_epi32(_First, _Second);
+    }
+};
+
+struct _Minmax_traits_8 {
+    using _Signed_t   = int64_t;
+    using _Unsigned_t = uint64_t;
+
+    static constexpr bool _Has_portion_max = false;
+
+    static constexpr uint64_t _Init_min_val = 0x7FFF'FFFF'FFFF'FFFFULL;
+    static constexpr uint64_t _Init_max_val = 0x8000'0000'0000'0000ULL;
+
+    static __m128i _Sign_cor(__m128i _Val, bool _Sign) {
+        alignas(16) static constexpr uint64_t _Sign_cors[2][2] = {
+            0x8000'0000'0000'0000ULL, 0x8000'0000'0000'0000ULL, {}};
+        return _mm_sub_epi64(_Val, _mm_loadu_si128(reinterpret_cast<const __m128i*>(_Sign_cors[_Sign])));
+    }
+
+    static __m128i _Inc() {
+        return _mm_set1_epi64x(1);
+    }
+
+    static __m128i _H_min(__m128i _Cur) {
+        int64_t _H_min_a = _mm_cvtsi128_si64(_Cur);
+        int64_t _H_min_b = _mm_cvtsi128_si64(_mm_bsrli_si128(_Cur, 8));
+        if (_H_min_b < _H_min_a) {
+            _H_min_a = _H_min_b;
+        }
+        return _mm_set1_epi64x(_H_min_a);
+    }
+
+    static __m128i _H_max(__m128i _Cur) {
+        int64_t _H_max_a = _mm_cvtsi128_si64(_Cur);
+        int64_t _H_max_b = _mm_cvtsi128_si64(_mm_bsrli_si128(_Cur, 8));
+        if (_H_max_b > _H_max_a) {
+            _H_max_a = _H_max_b;
+        }
+        return _mm_set1_epi64x(_H_max_a);
+    }
+
+    static int64_t _Get_any(__m128i _Cur) {
+        return static_cast<uint64_t>(_mm_cvtsi128_si64(_Cur));
+    }
+
+    static uint64_t _Get_v_pos(__m128i _Idx, unsigned long _H_pos) {
+        uint64_t _Array[2];
+        _mm_storeu_si128(reinterpret_cast<__m128i*>(&_Array), _Idx);
+        return _Array[_H_pos >> 3];
+    }
+
+    static __m128i _Cmp_eq(__m128i _First, __m128i _Second) {
+        return _mm_cmpeq_epi64(_First, _Second);
+    }
+
+    static __m128i _Cmp_gt(__m128i _First, __m128i _Second) {
+        return _mm_cmpgt_epi64(_First, _Second);
+    }
+
+    static __m128i _Cmp_lt(__m128i _First, __m128i _Second) {
+        __m128i _Gt = _mm_cmpgt_epi64(_Second, _First); // less or equal
+        __m128i _Eq = _mm_cmpeq_epi64(_First, _Second);
+        return _mm_andnot_si128(_Eq, _Gt);
+    }
+
+    static __m128i _Min(__m128i _First, __m128i _Second, __m128i _Mask) {
+        return _mm_blendv_epi8(_First, _Second, _Mask);
+    }
+
+    static __m128i _Max(__m128i _First, __m128i _Second, __m128i _Mask) {
+        return _mm_blendv_epi8(_First, _Second, _Mask);
+    }
+};
+
+template <_Min_max_mode _Mode, class _Traits>
+auto _Minmax_element(const void* _First, const void* _Last, bool _Sign) noexcept {
+    _Min_max_t _Res                          = {_First, _First};
+    auto _Base                               = static_cast<const _Traits::_Unsigned_t*>(_First);
+    typename _Traits::_Signed_t _Cur_min_val = _Traits::_Init_min_val;
+    typename _Traits::_Signed_t _Cur_max_val = _Traits::_Init_max_val;
+
+    if (_Byte_length(_First, _Last) >= 16 && _Use_sse42()) {

```

Comment:
Can we please make this an early return

Reply:
No. there's inlined tail call to tail processing, which is also fallback processing. This is similar to existing vector algorithm.